### PR TITLE
[SYCLomatic] Support EMU migration of 40 math APIs which were migrated to math::ext APIs.

### PR DESCRIPTION
**Pull request by tangjj11**
1. Add math related help function header file.
2. Add EMU migration rules of 11 half related APIs.
3. Add EMU migration rules of 29 half2 related APIs.
4. Format the APINamesMathRewrite.inc file.

Signed-off-by: Tang, Jiajun jiajun.tang@intel.com

### CODE REVIEW

File: clang/lib/DPCT/CallExprRewriter.h [hunk 1]
```diff
@@ -942,6 +942,37 @@ class BinaryOperatorPrinter {
   }
 };
 
+template <class CValueT, class LValueT, class RValueT>
+class TernaryOperatorPrinter {
```
Comment (zhiweij1) [inline clang/lib/DPCT/CallExprRewriter.h]: ConditionalOperatorPrinter ?

**Review by zhiweij1 — APPROVED** [review]

File: clang/lib/DPCT/CallExprRewriter.h (hunk 1 above)
Comment (tangjj11) [reply clang/lib/DPCT/CallExprRewriter.h]: deleted because use help function.

File: clang/runtime/dpct-rt/include/util.hpp.inc [hunk 2]
```diff
@@ -161,230 +161,6 @@ inline void matrix_mem_copy(T *to_ptr, const T *from_ptr, int to_ld,
 }
 // DPCT_LABEL_END
 
-// DPCT_LABEL_BEGIN|cast_double_to_int|dpct
```
Comment (zhiweij1) [inline clang/runtime/dpct-rt/include/util.hpp.inc:164]: How about:
1. add `namespace math` for all functions in math.hpp
2. keep interfaces here to call dpct::math::...

**Review by zhiweij1 — APPROVED** [review]

**Review by ziranzha — APPROVED** [review]

**Review by AndyCHHuang — APPROVED** [review]
LGTM

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**